source/updateCheck.py

```diff
@@ -237,28 +237,50 @@ def executePendingUpdate():
 		_executeUpdate(updateTuple[0])
 
 
-def _executeUpdate(destPath):
+def _executeUpdate(destPath: str) -> None:
+	"""Execute the update process.
+
+	:param destPath: The path to the update executable.
+	:raises ValueError: If destPath is empty or not given.
+	"""
 	if not destPath:
-		return
+		raise ValueError("destPath must be a non-empty string")
 
 	_setStateToNone(state)
 	saveState()
+	# #4475: ensure that the new process shows its first window, by providing SW_SHOWNORMAL
+	if not core.triggerNVDAExit(core.NewNVDAInstance(destPath, _generate_updateParameters())):
+		log.error("NVDA already in process of exiting, this indicates a logic error.")
+
+
+def _generate_updateParameters() -> str:
+	"""Generate parameters to pass to the new NVDA instance for the update process.
+
+	We generate parameters that specify:
+	- Whether to install, update a portable copy, or run the launcher.
+	- Whether to disable addons.
+	- The path to the configuration directory.
+
+	:return: The parameters to pass to the new NVDA instance.
+	"""
+	executeFlags: set[str] = set()
+	executeOptions: dict[str, str] = {}
 	if config.isInstalledCopy():
-		executeParams = "--install -m"
+		executeFlags.update(("--install", "-m"))
 	else:
 		portablePath = globalVars.appDir
 		if os.access(portablePath, os.W_OK):
-			executeParams = (
-				'--create-portable --portable-path "{portablePath}" --config-path "{configPath}" -m'.format(
-					portablePath=portablePath,
-					configPath=WritePaths.configDir,
-				)
-			)
+			executeFlags.update(("--create-portable", "-m"))
+			executeOptions["--portable-path"] = f'"{portablePath}"'
```

Comment:
```suggestion
			executeOptions["--portable-path"] = portablePath
```

Reply:
This will fail if there are spaces in `portablePath`.